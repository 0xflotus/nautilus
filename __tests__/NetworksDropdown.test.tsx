import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NetworksDropDown from '../src/renderer/components/NetworksDropdown';

configure({ adapter: new Adapter() });

const props = {
  networks: {
    hello: 'test1',
    yellow: 'test2',
    mellow: 'test3',
  },
  selectNetwork: jest.fn(() => {}),
  selectedNetwork: '',
};

describe('Test Networks Dropdown Component', () => {
  let wrapper: any;

  beforeAll(() => {
    wrapper = shallow(<NetworksDropDown {...props} />);
  });

  it('should have a disabled networks option', () => {
    expect(wrapper.find('#networkHeader')).toBeDisabled();
  });

  // it('should render as many options as there are networks, except if there are 1 or less networks in networks object', () => {
  //   const networksLength = Object.keys(props.networks).length;
  //   if (networksLength <= 1) {
  //     expect(wrapper.find('.networkOption')).toHaveLength(2);
  //   } else {
  //     expect(wrapper.find('.networkOption')).toHaveLength(networksLength + 2);
  //   }
  // });
});
