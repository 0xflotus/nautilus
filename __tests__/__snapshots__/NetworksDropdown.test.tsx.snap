// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Networks Dropdown Component renders correctly 1`] = `
<select
  className="option"
  id="networks"
  name="networks"
  onChange={[Function]}
  value=""
>
  <option
    disabled={true}
    id="networkHeader"
    value=""
  >
    networks
  </option>
  <option
    className="networkOption"
    id="snapTester"
    value="snapTester"
  >
    snapTester
  </option>
  <option
    className="networkOption"
    id="dbTester"
    value="dbTester"
  >
    dbTester
  </option>
  <option
    className="networkOption"
    id="groupNetworks"
    value="groupNetworks"
  >
    group networks
  </option>
</select>
`;
